pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -72,6 +72,21 @@ func (pr *PipelineRun) GetTaskRunRef() corev1.ObjectReference {
 	}
 }
 
+// GetTypeMeta returns the task run type meta
+func (pr *PipelineRun) GetTypeMeta() *metav1.TypeMeta {
+	return &pr.TypeMeta
+}
+
+// GetObjectMeta returns the task run type meta
+func (pr *PipelineRun) GetObjectMeta() *metav1.ObjectMeta {
+	return &pr.ObjectMeta
+}
+
+// GetStatus returns the task run status as a RunsToCompletionStatus
+func (pr *PipelineRun) GetStatus() RunsToCompletionStatus {
+	return &pr.Status
+}
+
```

Comment:
What is the benefit with adding these methods, compared to use the fields directly? e.g. `pr.TypeMeta`, `pr.ObjectMeta` and `pr.Status` ?

Update: Now, I understand - it is part of an interface. Maybe it doesn't have to be part of the interface? I don't know.

Reply:
Indeed, it's part of the interface. The idea is that a `RunsToCompletion` has a `RunsToCompletionStatus` from which Conditions can be extracted